File: exercises/advent-of-code/2021/clj/src/day2/part1.clj
(ns day2.part1
  (:require
   [clojure.string :as s]
   [parser :as parser]))

(defn parse-line
  [line]
  (let [[move amount] (s/split line #" ")]
    [move (Integer/parseInt amount)]))

(def input
  (parser/input parse-line "data/day2.txt"))

(defn step
  [[position depth]
   [move amount]]
  (case move
    "forward" [(+ position amount) depth]
    "down" [position (+ depth amount)]
    "up" [position (- depth amount)]))

(def result
  (reduce step [0 0] input))

(def result
  (apply * result))

(println result)

File: exercises/advent-of-code/2021/python/src/day6/part1.py
def parse(filename: str) -> list[int]:
    with open(filename) as file:
        line = file.read()
        return list(map(int, line.split(',')))

def solve(fishes: list[int], days: int) -> int:
    for _ in range(days):
        for i in range(len(fishes)):
            fishes[i] -= 1
            if fishes[i] < 0:
                fishes[i] = 6
                fishes.append(8)

    return len(fishes)

def main():
    input = parse('data/day6.txt')
    result = solve(input, 80)
    print(result)

if __name__ == '__main__':
    main()